snpe/tutorials/marketplace_simulation.ipynb

```diff
@@ -0,0 +1,957 @@
+       "                var nbb_cells = Jupyter.notebook.get_cells();\n",
+       "                for (var i = 0; i < nbb_cells.length; ++i) {\n",
+       "                    if (nbb_cells[i].input_prompt_number == nbb_cell_id) {\n",
+       "                        if (nbb_cells[i].get_text() == nbb_unformatted_code) {\n",
+       "                             nbb_cells[i].set_text(nbb_formatted_code);\n",
+       "                        }\n",
+       "                        break;\n",
+       "                    }\n",
+       "                }\n",
+       "            }, 500);\n",
+       "            "
+      ],
+      "text/plain": [
+       "<IPython.core.display.Javascript object>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "def input_generator(\n",
+    "    number_prods: int, number_users: int, artifact_path=\"../artifacts/marketplace/ \"\n",
+    "):\n",
+    "    \"\"\"\n",
+    "    This function generates and saves fictional inputs for a marketplace simulation. \n",
+    "    It requires the number of products and users to include in the fictional inputs as arguments. \n",
+    "    Output is stored by default in the path \"snpe/artifacts/marketplace/\", \n",
+    "    which is the path where the instances of the marketplace simulation-related classes are instructed \n",
+    "    to look for the inputs by default.\n",
+    "    A different path storage can be provided as the value of the optional \"artifact_path\" argument. \n",
+    "    \"\"\"\n",
+    "    assert (\n",
+    "        number_prods < 13939\n",
+    "    ), \"Argument number_user cannot be set at a avalue higher than 13939\"\n",
+    "    ids_alone = []\n",
+    "    ids_product = []\n",
+    "\n",
+    "    # Generating a set of random product ids to be used in rating histograms and embeddings\n",
+    "    while len(ids_alone) < number_prods:\n",
+    "        a = random.randint(1000000, 9999999)\n",
+    "        if a not in ids_alone:\n",
+    "            ids_alone.append(a)\n",
+    "            ids_product.append(\"product_\" + str(a))\n",
+    "\n",
+    "    # Generating 100d arrays to be used as product embeddings\n",
```

Comment:
Once again use numpy :) 
To generate 100-D arrays of random numbers, you can just do something like:
`np.random.normal(size=(number_prods, 100))`
Drawing random numbers from the normal distribution here is more appropriate than uniform (which is what random.random does) as these embeddings have values in (-inf, +inf) and not in [0,1] like uniform.

Reply:
Got it!